genesiscommunitysuccess/docs
# feat(docs): rollup script and config for generating iife files PTL-703 PTL-720

Jira tickets:
PTL-703 PTL-720

How to test:
Run:
```
git reset --hard && git clean --force -dfx && npm cache clean --force
npm i
npm run build:next
npm run serve
```
Then check the design system page (/web/design-systems/preview/)

# Notes:

We have tried disabling webpack optimizations (created a custom plugin for this) without much luck.
```
 optimization: {
      mangleExports: false,
      minimize: false,
      splitChunks: false,
      moduleIds: 'deterministic'
}
```
Something still breaks, likely webpack bundling related.

The current workaround is to manually copy the generated iife bundle into node_modules and pretend it is an external dependency. Ideally we want to clean this up in the future. 

I have added a bunch of iife related scripts to the postinstall hook so it should be transparent to devs working on docs / anyone performing upgrades. After every `npm i` a new iife bundle will be generated and placed into node_modules etc.
